providers/implementations/storemgmt/file_store.c

```diff
@@ -434,71 +430,49 @@ static int file_setup_decoders(struct file_ctx_st *ctx)
             goto err;
         }
 
-        expect_evp_pkey = (ctx->expected_type == 0
-                           || ctx->expected_type == OSSL_STORE_INFO_PARAMS
-                           || ctx->expected_type == OSSL_STORE_INFO_PUBKEY
-                           || ctx->expected_type == OSSL_STORE_INFO_PKEY);
-
         /* Make sure the input type is set */
         if (!OSSL_DECODER_CTX_set_input_type(ctx->_.file.decoderctx,
                                              ctx->_.file.input_type)) {
             ERR_raise(ERR_LIB_PROV, ERR_R_OSSL_DECODER_LIB);
             goto err;
         }
 
-        /*
-         * Create the internal last resort decoder implementation together
-         * with a "decoder instance".
-         * The decoder doesn't need any identification or to be attached to
-         * any provider, since it's only used locally.
-         */
-        to_obj = ossl_decoder_from_algorithm(0, &ossl_der_to_obj_algorithm,
-                                             NULL);
-        if (to_obj == NULL)
-            goto err;
-        to_obj_inst = ossl_decoder_instance_new(to_obj, ctx->provctx);
-        if (to_obj_inst == NULL)
-            goto err;
+        for (to_algo = ossl_any_to_obj_algorithm;
+             to_algo->algorithm_names != NULL;
+             to_algo++) {
+            OSSL_DECODER *to_obj = NULL;
```

Comment:
This initialization does not seem necessary as you always assign a value to the variable. It might be flagged by some compilers/static analyzers.

Reply:
> It might be flagged by some compilers/static analyzers.

Srsly????????
I have seen them flag for under-initializing, for good reason...  but for over-intializing???
Pardon me, but I remain doubtful.  Let's wait 'til some analyzer complains, yeah?  Feel free to shame me if that happens.

Reply:
There are definitely analyzers that complain about "dead store"s.
I would *hope* that they are smart enough to tolerate unused initializations at declaration, since there are good robustness and style reasons to do that, even if the initialization is not actually used right now (but might after some future refactoring/etc.).
(I'm fine with "wait and see", to be clear.)